assets/src/edit-story/app/media/useUploadMedia.js

```diff
@@ -170,6 +170,10 @@ function useUploadMedia({
             'File could not be uploaded. Please try a different file.',
             'web-stories'
           ),
+        thumbnail: {
+          src: resource.type === 'video' ? resource.poster : resource.src,
```

Comment:
Might just be a better to check if poster is set and if it is use it and then fallback to src. Worth noting that as of #7232 the resource type gif will also have a poster.

Reply:
> Might just be a better to check if poster is set and if it is use it and then fallback to src.

That would mean if a video does not have a poster, the `src` (the actual video) would be used as a thumbnail here. Not sure if that is desired.

Reply:
I will stick with the explicit type check for now for using `poster` and if there is no resource show nothing for now